src/psyclone/psyir/nodes/node.py

```diff
@@ -329,7 +330,19 @@ def coloured_name(self, colour=True):
                 "given a string value in the concrete class '{0}'."
                 "".format(type(self).__name__))
         if colour:
-            return colored(self._text_name, self._colour)
+            if self._colour is None:
+                raise NotImplementedError(
+                    "The _colour attribute is abstract so needs to be given "
+                    "a string value in the concrete class '{0}'."
+                    "".format(type(self).__name__))
+            try:
+                return colored(self._text_name, self._colour)
+            except KeyError as info:
+                message = (
+                    "The _colour attribute in class '{0}' has been set to an "
```

Comment:
"...set to a colour ('{0}') that is not supported by the termcolor package."

Reply:
Yes, that's better, thanks.